### ISSUE TITLE
path/filepath: Glob with UNC path causes stack overflow

### ISSUE THREAD

**Issue by AudriusButkevicius** [golang/go#15879]
1. What version of Go are you using (`go version`)?

`go version devel +b859a78 Sun May 29 06:52:45 2016 +0000 windows/amd64`

2. What operating system and processor architecture are you using (`go env`)?

```
set GOARCH=amd64
set GOBIN=
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\GoHome
set GORACE=
set GOROOT=C:\Go
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set CC=gcc
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0
set CXX=g++
set CGO_ENABLED=1
```

3. What did you do?

```
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
   res, _ := filepath.Glob(`\\?\C:\*`)
    for _, match := range res {
        fmt.Println(match)
    }
}
```

4. What did you expect to see?

Nothing (due to UNC path) or listing of C drive

5. What did you see instead?

```
runtime: goroutine stack exceeds 1000000000-byte limit
fatal error: stack overflow

runtime stack:
runtime.throw(0x4cdf36, 0xe)
	C:/Go/src/runtime/panic.go:566 +0x9c
runtime.newstack()
	C:/Go/src/runtime/stack.go:1054 +0x424
runtime.morestack()
	C:/Go/src/runtime/asm_amd64.s:366 +0x87

goroutine 1 [running]:
strings.ContainsAny(0x4ccd09, 0x7, 0x4cc5f0, 0x3, 0x483c20)
	C:/Go/src/strings/strings.go:130 fp=0xc0620712d0 sp=0xc0620712c8
path/filepath.hasMeta(0x4ccd09, 0x7, 0x3f)
	C:/Go/src/path/filepath/match.go:334 +0x51 fp=0xc062071308 sp=0xc0620712d0
path/filepath.Glob(0x4ccd09, 0x7, 0x4ccd01, 0x7, 0x4ccd09, 0x0, 0xc062071420)
	C:/Go/src/path/filepath/match.go:235 +0x54 fp=0xc0620713b8 sp=0xc062071308
path/filepath.Glob(0x4ccd09, 0x7, 0x4ccd01, 0x7, 0x4ccd09, 0x0, 0xc0620714d0)
	C:/Go/src/path/filepath/match.go:254 +0xfb fp=0xc062071468 sp=0xc0620713b8
path/filepath.Glob(0x4ccd09, 0x7, 0x4ccd01, 0x7, 0x4ccd09, 0x0, 0xc062071580)
	C:/Go/src/path/filepath/match.go:254 +0xfb fp=0xc062071518 sp=0xc062071468
...
```

**Comment by alexbrainman** [comment]
I am travelling until mid June. I won't be able to fix this until I am back. Unless someone else wants to fix this.

I didn't consider UNC for filepath.Glob. Given that we use \ to escape in filepath.Glob, it could be quite invasive to make UNCs work.

Alex

**Comment by AudriusButkevicius** [comment]
I think it's sufficient just to strip the prefix if it exists, as Glob walks directory by directory anyway.

**Comment by josharian** [comment]
The stack overflow seems bad enough to fix for 1.7. I have a minimal, safe fix prepared that returns ErrBadPattern for this input instead of crashing. We could make a better fix for 1.8 that would return files instead of an error. What do you think? @adg, thoughts about milestone?

**Comment by ianlancetaylor** [comment]
@josharian If you have a fix, let's get it in for 1.7.

**Comment by josharian** [comment]
CL 23572

**Comment by adg** [comment]
Changed the milestone to 1.8 now that a fix is in for 1.7.
